trunner/testcase.py

```diff
@@ -171,6 +188,11 @@ def handle(self, proc):
             msg = 'EXCEPTION ' + ('EOF' if isinstance(exc, EOF) else 'TIMEOUT') + '\n'
             self.exception += Color.colorify(msg, Color.BOLD)
             self.handle_pyexpect_error(proc, exc)
+        except UnicodeDecodeError:
+            msg = 'Unicode Decode Error detected!\n'
+            self.exception = Color.colorify(msg, Color.BOLD)
+            self.handle_unicode_error(proc)
+            return
```

Comment:
I see a lot of code duplication in exception handling here. Those exact 4 lines appear 3 times. Could this try block be extended to unify catching errors from both harness and exec test?

Reply:
Yeah, I've checked this and it can be done in one try block here. I would like to leave it as it is in handle function.